src/Illuminate/Routing/AbstractRouteCollection.php

```diff
@@ -101,24 +104,26 @@ protected function getRouteForMethods($request, array $methods)
             }))->bind($request);
         }
 
-        $this->methodNotAllowed($methods, $request->method());
+        $this->methodNotAllowed($request, $methods, $request->method());
     }
 
     /**
      * Throw a method not allowed HTTP exception.
      *
+     * @param  \Illuminate\Http\Request  $request
      * @param  array  $others
      * @param  string  $method
      * @return void
      *
      * @throws \Symfony\Component\HttpKernel\Exception\MethodNotAllowedHttpException
      */
-    protected function methodNotAllowed(array $others, $method)
+    protected function methodNotAllowed($request, array $others, $method)
```

Comment:
This is a breaking change to the method signature.

Reply:
We can have two of them. One for the old behaviour and one for the new behaviour.